### ISSUE TITLE
Wrong SQL query generated for left_join() when column names same but case

### ISSUE THREAD

**Issue by kelnerrr** [sparklyr/sparklyr#3219]
I encountered a bug when two tables have same joining column names that differ only in case (e.g. "x" and "X").
When they are the same the query is correct:
```
temp_tibble1 <- tibble(x = 1:5, y = 11:15) %>% copy_to(sc, ., name = "temp_tibble1", overwrite = TRUE)
temp_tibble2 <- tibble(x = 1:5, z = 111:115) %>% copy_to(sc, ., "temp_tibble2", overwrite = TRUE)

temp_tibble1 %>%
  left_join(temp_tibble2, by = c("x" = "x")) %>% show_query()
```
giving

```
<SQL>
SELECT `LHS`.`x` AS `x`, `y`, `z`
FROM `temp_tibble1` AS `LHS`
LEFT JOIN `temp_tibble2` AS `RHS`
ON (`LHS`.`x` = `RHS`.`x`)
```
but
```
temp_tibble1 <- tibble(x = 1:5, y = 11:15) %>% copy_to(sc, ., name = "temp_tibble1", overwrite = TRUE)
temp_tibble2 <- tibble(X = 1:5, z = 111:115) %>% copy_to(sc, ., "temp_tibble2", overwrite = TRUE)

temp_tibble1 %>%
  left_join(temp_tibble2, by = c("x" = "X")) %>% show_query()
```
gives
```
<SQL>
SELECT `x`, `y`, `z`
FROM `temp_tibble1` AS `LHS`
LEFT JOIN `temp_tibble2` AS `RHS`
ON (`LHS`.`x` = `RHS`.`X`)
```
which fails with

> Error : org.apache.spark.sql.AnalysisException: Reference 'x' is ambiguous, could be: LHS.x, RHS.x.; line 1 pos 7

**Comment by kelnerrr** [comment]
I reported an issue on dbplyr's page (#767) as I think the problem is there. For now I leave the issue here waiting for confirmation.

**Comment by kelnerrr** [comment]
Closing as I got the confirmation:
*I close it because it is already fixed in the dev version (see #716).
You could install the dev version devtools::install_github("tidyverse/dbplyr") or wait until the next release.*
